File: C#/SDK1/TExcel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Excel = Microsoft.Office.Interop.Excel;
using System.Reflection;


namespace SDK
{
    public partial class TExcel_frm : Form
    {
        Excel.Application oExcel;
        Excel._Workbook oWBook;
        Excel._Worksheet oSheet;
        PlanSwift.Tab CurrentTab;
        int i = 2;
        public TExcel_frm()
        {
            InitializeComponent();
        }
        
        public void SendToExcel(PlanSwift.Item AItem)
        {
            PlanSwift.ItemType IType;
            object ItemProp, IName, IQty, ITab, INumber, IPriceEach, IPriceTotal;
            IType = AItem.ItemType;
            ItemProp = AItem.Properties;
            ITab = TGlobal.pc.Tabs.SelectedTab().Name;
            IName = AItem.Name;
            if (AItem.ItemType == PlanSwift.ItemType.itFolder)
            {
                INumber = "";
                IQty = "";
                IPriceEach = "";
                IPriceTotal = "";
            }
            else
            {
                INumber = AItem.Properties.ByName("Item #").Value();
                IQty = AItem.Properties.ByName("Qty").Value();
                IPriceEach = AItem.Properties.ByName("Price Each").Value();
                IPriceTotal = AItem.Properties.ByName("Price Total").Value();    
            }
            oSheet.Cells[i, 1] = ITab;
            oSheet.Cells[i, 2] = IName;
            oSheet.Cells[i, 3] = INumber;
            oSheet.Cells[i, 4] = IQty;
            oSheet.Cells[i, 5] = IPriceEach;
            oSheet.Cells[i, 6] = IPriceTotal;
            switch (IType)
            {
                case PlanSwift.ItemType.itNone:
                    oSheet.Cells[i, 7] = "";
                    break;
                case PlanSwift.ItemType.itPart:
                    oSheet.Cells[i, 7] = "Part";
                    break;
                case PlanSwift.ItemType.itAssembly:
                    oSheet.Cells[i, 7] = "Assembly";
                    break;
                case PlanSwift.ItemType.itFolder:
                    oSheet.Cells[i, 7] = "Folder";
                    break;
            }
            i++;
        }
        public void GetItems(PlanSwift.Item AItem)
        {
            SendToExcel(AItem);
            for (int ChildIndex = 0; ChildIndex <= AItem.Count - 1; ChildIndex++ )
            {
                GetItems(AItem[ChildIndex]);
            }
        }
        private void TExcel_frm_Load(object sender, EventArgs e)
        {
            for (int TabIndex = 0; TabIndex <= TGlobal.pc.Tabs.Count -1; TabIndex++)
            {
                if (TGlobal.pc.Tabs[TabIndex].TabType == PlanSwift.TabType.ttJob)
                {
                    Tabs_cbx.Items.Add(TGlobal.pc.Tabs[TabIndex].Name);
                }
            }
        }

        private void button1_Click(object sender, EventArgs e)
        {
            if (Tabs_cbx.Text == "")
            {
                MessageBox.Show("Please Select a Tab", "Select a Tab", MessageBoxButtons.OK, MessageBoxIcon.Error);
                Tabs_cbx.Focus();
                return; 
            }
            //Excel.Application oExcel;
            //Excel._Workbook oWBook;
            //Excel._Worksheet oSheet;
            //PlanSwift.Tab CurrentTab;
            int i = 0;
            // Start Excel and get Application object
            oExcel = new Excel.Application();
            
            // Create new Workbook
            oWBook = (Excel._Workbook)(oExcel.Workbooks.Add(Missing.Value));
            oSheet = (Excel._Worksheet)oWBook.ActiveSheet;
            // Add Header Information
            oSheet.Cells[1, 1] = "Tab";
            oSheet.Cells[1, 2] = "Name";
            oSheet.Cells[1, 3] = "Item Number";
            oSheet.Cells[1, 4] = "Qty";
            oSheet.Cells[1, 5] = "Price Each";
            oSheet.Cells[1, 6] = "Price Total";
            oSheet.Cells[1, 7] = "Item Type";
            // Apply Header Formating
            oSheet.get_Range("A1", "G1").Font.Bold = true;
            oSheet.get_Range("A1", "G1").VerticalAlignment = Excel.XlVAlign.xlVAlignCenter;

            //Get Selected Planswift Tab
            CurrentTab = TGlobal.pc.Tabs.ByName(Tabs_cbx.Text);
            CurrentTab.MakeActive();
            for (int ItemIndex = 0; ItemIndex <= CurrentTab.Count - 1; ItemIndex++)
            {
                GetItems(CurrentTab[ItemIndex]);
                i = ItemIndex;
            }
            oSheet.get_Range("A1", "G1").EntireColumn.AutoFit();
            oExcel.Visible = true;
            oExcel = null;
            oWBook = null;
            oSheet = null;
        }
    }
}

File: Visual Basic/SDK Ver1.5/TExcel.vb
Public Class TExcel_frm
  Dim Excel As Microsoft.Office.Interop.Excel.Application
'**********************************EXPORT TO EXCEL*********************************************
  Dim i As Integer = 2
  Dim didcreate As Boolean 

  Sub SendToExcel(ByVal AItem As PlanSwift.Item)' Sub Funciton
    'Send Items to Excel
    Dim IType As planswift.ItemType
    Dim ItemProp, IName, IQty,ITab,INumber,IPriceEach, IPriceTotal
    
    ItemProp = AItem.Properties
    ITab = pc.Tabs.SelectedTab.Title
    IName = AItem.Name
    If AItem.Type_ = PlanSwift.ItemType.itFolder then
      INumber = ""
      IQty = ""
      IPriceEach = ""
      IPriceTotal = ""
    Else
      INumber = AItem.Properties.ByName("Item #").Value
      IQty = AItem.Properties.ByName("Qty").Value
      IPriceEach = AItem.Properties.ByName("Price Each").Value
      IPriceTotal = AItem.Properties.ByName("Price Total").Value
    End If
    
    IType = AItem.Type_
    Excel.Cells(i,1).Value = ITab
    Excel.Cells(i,2).Value = IName
    Excel.Cells(i,3).Value = INumber
    Excel.Cells(i,4).Value = IQty
    Excel.Cells(i,5).Value = IPriceEach
    Excel.Cells(i,6).Value = IPriceTotal 
    Select Case Itype
      Case 0 
       Excel.Cells(i,7).Value = ""
      Case 1 
       Excel.Cells(i,7).Value = "Part"
      Case 2
       Excel.Cells(i,7).Value = "Assembly"
      Case 3
        Excel.Cells(i,7).Value = "Folder"
    End Select
     i = i +1 
  End Sub
sub GetItems(ByVal AItem As PlanSwift.Item)
'Gather all planswift items  
SendToExcel(AItem)
  For ChildIndex = 0 to AItem.Count - 1
    GetItems(AItem.Item(ChildIndex))
  Next
End Sub
 
Private Sub Form1_Load( ByVal sender As System.Object,  ByVal e As System.EventArgs) Handles MyBase.Load    

  Dim TabIndex As Integer 
  'Add Planswift tabs to combobox   
  For TabIndex = 0 to pc.Tabs.Count -1 
    If pc.Tabs(TabIndex).TabType = PlanSwift.TabType.ttJob Then
      Tabs_cbx.Items.Add(pc.Tabs(TabIndex).Name)
    End If
  Next
End Sub

Private Sub ok_btn_Click( ByVal sender As System.Object,  ByVal e As System.EventArgs) Handles ok_btn.Click 
 If Tabs_cbx.Text = "" Then
      MessageBox.Show("Please Select a Tab","Select a Tab",MessageBoxButtons.OK,MessageBoxIcon.Error)
      Tabs_cbx.Focus
      Exit Sub
    End If
Dim CurrentTab As PlanSwift.Tab
Dim ItemIndex As Integer
Dim ExWb,ExWSheet, ExRng
    'Create Excel Object
    on Error Resume Next
    Excel = GetObject("Excel.Application")
    If Err.Description > vbNullString Then
      Err.Clear
      Excel = CreateObject("Excel.Application")
      If Err.Description > vbNullString Then
        MsgBox("Application Not Found",MsgBoxStyle.Critical,"Error!")
        Me.Close
        Exit Sub  
      End If
    End If
      Excel.Visible = True
      ExWb = Excel.Workbooks.Add
      ExWSheet = Excel.ActiveSheet
   
    'Add Header information
    ExWSheet.Cells(1,1).value = "Tab"
    ExWSheet.Cells(1,2).Value = "Name"
    ExWSheet.Cells(1,3).Value = "Item Number"
    ExWSheet.Cells(1,4).Value = "Qty"
    ExWSheet.Cells(1,5).Value = "Price Each"
    ExWSheet.Cells(1,6).Value = "Price Total"
    ExWSheet.Cells(1,7).Value = "Item Type"
   
   'Add Excel Formating
    With ExWSheet.Range("A1","G1")
      .Font.Bold = True  
    End With

    'Add Planswift items to excel Sheet
    CurrentTab = pc.Tabs.ByName(Tabs_cbx.Text)
    CurrentTab.MakeActive
    For ItemIndex = 0 To CurrentTab.Count - 1
      GetItems(CurrentTab(ItemIndex))
    next
    ExRng = ExWSheet.Range("A1","G" & i)
    ExRng.EntireColumn.Autofit  
    Excel = Nothing
    Exit Sub
    

End Sub
'***********************************END EXPORT TO EXCEL**********************************************

End Class
